src/body.rs

```diff
@@ -272,11 +272,15 @@ impl BodyReader {
         use crate::ext::MethodExt;
 
         let is_head = method == Method::HEAD;
-
         if is_head {
             return Ok(Self::NoBody);
         }
 
+        let is_connect = method == Method::CONNECT;
+        if is_connect {
+            return Ok(Self::NoBody);
+        }
```

Comment:
If you search for the trait `MethodExt`, we could potentially simplify some code here by expanding on that idea.